mailpoet/lib/Subscribers/SubscriberSegmentRepository.php

```diff
@@ -81,8 +81,34 @@ public function unsubscribeFromSegments(SubscriberEntity $subscriber, array $seg
   }
 
   public function resetSubscriptions(SubscriberEntity $subscriber, array $segments): void {
-    $this->unsubscribeFromSegments($subscriber);
-    $this->subscribeToSegments($subscriber, $segments);
+    $existingSegments = array_values(array_filter(array_map(
```

Comment:
Please, could you add some comments here? It could help to better and faster understand how it works.

Reply:
a19774de7